src/main/java/UseCase/BuildMatrix.java

```diff
@@ -7,54 +7,52 @@
 
 public class BuildMatrix {
 
-    private int rows;
-    private int cols;
+    public Matrix createMatrix(){
 
 
-    public int getRows() {
-        return rows;
-    }
+        Scanner input = new Scanner(System.in);
 
-    public void setRows(int rows) {
-        this.rows = rows;
-    }
+        System.out.println("Please type a square matrix using brackets: ");
 
-    public int getCols() {
-        return cols;
-    }
+        String clean = input.nextLine().replaceAll(" ", "");
 
-    public void setCols(int cols) {
-        this.cols = cols;
-    }
+        String[] matrix = clean.split("");
 
-    public ArrayList<ArrayList<Fraction>> buildMatrix(){
-        Scanner input = new Scanner(System.in);
+        return new Matrix(helper(matrix));
 
 
-        System.out.println("Please enter the number of rows: ");
-        int rows = input.nextInt();
-        System.out.println("Please enter the number of columns: ");
-        int cols = input.nextInt();
-        System.out.println("Your matrix size is " + rows + " x " + cols);
+    }
 
-        ArrayList<ArrayList<Fraction>> matrix = new ArrayList<>();
+    public ArrayList<ArrayList<Fraction>> helper(String[] matrix){
+
+        int x  = 1;
+        ArrayList<ArrayList<Fraction>> matri = new ArrayList<>();
+        while (x < matrix.length){
+            if (Objects.equals(matrix[x].toString(), "[")){
+                x++;
+                ArrayList<Fraction> inner = new ArrayList<>();
+                while (!matrix[x].toString().equals("]")){
```

Comment:
if you're going to use 'matrix[x].toString()' multiple times make it a variable

Reply:
I will look into this later as the x value changes